azurerm/internal/services/resource/tenant_template_deployment_resource.go

```diff
@@ -0,0 +1,369 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func tenantTemplateDeploymentResource() *schema.Resource {
+	return &schema.Resource{
+		Create: tenantTemplateDeploymentResourceCreate,
+		Read:   tenantTemplateDeploymentResourceRead,
+		Update: tenantTemplateDeploymentResourceUpdate,
+		Delete: tenantTemplateDeploymentResourceDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.TenantTemplateDeploymentID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(180 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(180 * time.Minute),
+			Delete: schema.DefaultTimeout(180 * time.Minute),
+		},
+
+		// (@jackofallops - lintignore needed as we need to make sure the JSON is usable in `output_content`)
+
+		//lintignore:S033
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.TemplateDeploymentName,
+			},
+
+			"location": location.Schema(),
+
+			"template_content": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ExactlyOneOf: []string{
+					"template_content",
+					"template_spec_version_id",
+				},
+				StateFunc: utils.NormalizeJson,
+			},
+
+			"template_spec_version_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ExactlyOneOf: []string{
+					"template_content",
+					"template_spec_version_id",
+				},
+				ValidateFunc: validate.TemplateSpecVersionID,
+			},
+
+			// Optional
+			"debug_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(templateDeploymentDebugLevels, false),
+			},
+
+			"parameters_content": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				StateFunc: utils.NormalizeJson,
+			},
+
+			"tags": tags.Schema(),
+
+			// Computed
+			"output_content": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				StateFunc: utils.NormalizeJson,
+				// NOTE:  outputs can be strings, ints, objects etc - whilst using a nested object was considered
+				// parsing the JSON using `jsondecode` allows the users to interact with/map objects as required
+			},
+		},
+	}
+}
+
+func tenantTemplateDeploymentResourceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Resource.DeploymentsClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewTenantTemplateDeploymentID(d.Get("name").(string))
```

Comment:
should this be taking the tenant Id here?

Reply:
The API doesn't take it, it assumes the tenant is that the requesting client is authenticated against.